TODO/object-detection-with-yolo.md

```diff
@@ -244,9 +255,10 @@ let tc = features[offset(channel + 4, cx, cy)]
 
 ```
 
-The `offset()` helper function is used to find the proper place in the array to read from. Metal stores its data in texture slices in groups of 4 channels at a time, which means the 125 channels are not stored consecutively but are scattered all over the place. (See the code for an in-depth explanation.)
 
-We still need to do some processing on these five numbers `tx`, `ty`, `tw`, `th`, `tc` as they are in a bit of a weird format. If you’re wondering where these formulas come from, they’re given [in the paper](https://arxiv.org/abs/1612.08242) (it’s a side effect of how the network was trained).
+帮助函数 `offset()` 用来定位数组中合适的读取位置。Metal 以每次 4 个通道一组来把数据存在纹理片中，这意味着 125 个通道不是连续存储，而是分散存储的。（想深入分析的话可以去看源码）。
+
+我们仍然需要处理 `tx`， `ty`， `tw`， `th`， `tc` 这五个参数 ，因为它们的格式有点奇怪。如果你不知道这些处理方法哪来的话，可以看下这篇[论文](https://arxiv.org/abs/1612.08242) (这是网络如何训练的副作用)。
```

Comment:
>  这是训练这个神经网络的附加产物之一 

仔细看了一下这篇论文就是 YOLO 的作者写的，所以这篇论文应该是对训练过程的一个更详细的描述。是不是训练的过程最后形成了这篇论文？🤔

YOLO 官网有这么个描述，这个 paper 是同一个 paper：
>What's New in Version 2?

YOLOv2 uses a few tricks to improve training and increase performance. Like Overfeat and SSD we use a fully-convolutional model, but we still train on whole images, not hard negatives. Like Faster R-CNN we adjust priors on bounding boxes instead of predicting the width and height outright. However, we still predict the x and y coordinates directly. The full details are in our paper.!